<?xml version="1.0"?>
<page id="Gio.Task.return_error"
      type="topic"
      style="method"
      xmlns="http://projectmallard.org/1.0/"
      xmlns:api="http://projectmallard.org/experimental/api/"
      xmlns:ui="http://projectmallard.org/1.0/ui/">
  <info>
    <link xref="Gio.Task" group="method" type="guide"/>
    <api:function>
      <api:returns>
        <api:type>void</api:type>
      </api:returns>
      <api:name>g_task_return_error</api:name>
      <api:arg>
        <api:type>GLib.Error</api:type>
        <api:name>error</api:name>
      </api:arg>
    </api:function>
  </info>
  <title>Gio.Task.prototype.return_error</title>
  <synopsis><code mime="text/x-gjs">
function return_error(error: <link href="../GLib-2.0/GLib.Error.html">GLib.Error</link>): void {
    // Gjs wrapper for g_task_return_error()
}
  </code></synopsis>
  <p>Sets <code>task</code>'s result to <code>error</code> (which <code>task</code> assumes ownership of)
and completes the task (see <link xref="Gio.Task.return_pointer"/> for more
discussion of exactly what this means).</p>  <p>Note that since the task takes ownership of <code>error</code>, and since the
task may be completed before returning from <link xref="Gio.Task.return_error"/>,
you cannot assume that <code>error</code> is still valid after calling this.
Call <link href="../GLib-2.0/GLib.Error.copy.html">GLib.Error.prototype.copy</link> on the error if you need to keep a local copy
as well.</p>  <p>See also <link xref="Gio.Task.return_new_error"/>.</p>
  <p>Since 2.36</p>
<terms>
<item>
<title><code>error</code></title>
  <p>the <link href="../GLib-2.0/GLib.Error.html">GLib.Error</link> result of a task function.</p>
</item>
</terms>

</page>
